### PR TITLE
Update root post update at on reply delete

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
When we create or update a post in a thread, the root post also gets their update at updated to the same value. But when a post is deleted, it is not.

This created an issue when checking the etag, which was looking for the latest update at, and if two are the same, the one with the "highest id". If any post "in the middle" or a last post with "lower id" is removed, the update at would be the same (the same as the last posted post) and the id too,generating the same etag and forcing this issue.

By updating the `update_at` of the post whenever we delete a post fixes the issue.

#### Ticket Link
Fix https://mattermost.atlassian.net/browse/MM-60791

#### Release Note
```release-note
Fix issue when deleted posts on threads may not appear as deleted on certain circumstances
```

### CODE REVIEW

File: server/channels/store/sqlstore/post_store.go [hunk 1]
```diff
@@ -942,6 +942,9 @@ func (s *SqlPostStore) Delete(rctx request.CTX, postID string, time int64, delet
 		err = s.deleteThread(transaction, postID, time)
 	} else {
 		err = s.updateThreadAfterReplyDeletion(transaction, id.RootId, id.UserId)
+		if _, err = s.GetMasterX().Exec("UPDATE Posts SET UpdateAt = ? WHERE Id = ?", time, id.RootId); err != nil {
+			mlog.Warn("Error updating Post UpdateAt.", mlog.Err(err))
```
Comment (cpoile) [inline server/channels/store/sqlstore/post_store.go:946]: Should be in the transaction, I think. And we should be using squirrel for new code.

Comment (larkox) [reply server/channels/store/sqlstore/post_store.go]: Added the operation to the transaction. Not sure if it has any potential implications if that particular part of the transaction fails, but probably is the best. That being said, when creating a post, a few things fall outside of the transaction, including updating the root post last post at, so not sure 100%.

Re: squirrel, I also updated it, but the code is not really "new". It was mostly a copy paste from what is used on save and update. I just considered creating a function like `updateThreadAfterReplyDeletion` to reuse in the three places it is used (create, update and delete) but the three of them handle transactions differently, so I guess such "unification" is better handled in a different PR. That being said, I am open to suggestions.

**Review by jespino — APPROVED** [review]
LGTM

**Review by cpoile — APPROVED** [review]
Awesome, thanks @larkox!

**Review by yasserfaraazkhan — APPROVED** [review]
threaded post dont have deleted post re-appearing when I close and open threads with CRT enabled

**Pull request merged**